docs/tech-notes/sql.md

```diff
@@ -442,197 +502,44 @@ Two things are involved here:
-
-CockroachDB currently implements two back-ends.
-
-- the **local runner** is used by default, and always for statements
-  that don't have a distributed implementation (e.g. CREATE TABLE,
-  COPY).
-- the **distSQL back-end** is used when `SET DISTSQL` is set to
-  `auto`, `on` or `always`.
-
-### Local runner
-
-**Role:** run SQL statements on the SQL gateway node (the node where
-pgwire is handling the client connection).
-
-**Interface:**
-
-- interface from the perspective of the executor:
-  - conceptually: provide a logical plan as input, get row sets (or
-    error status) as output.
-  - concretely: the `Start()` and `Next()` methods of `planNode`.
-- lower-level interface:
-  - the **sql schema changer** for DDL statements;
-  - the **client KV interface** for "leaf" plan nodes that access
-    the database.
-- towards the memory monitor: to account for intermediate results
-  held in memory.
-
-**How:**
-
-- uses the Volcano execution model - see https://doi.org/10.1109/69.273032
-  - each level's `Next()` recurses into sub-nodes to request rows, and returns
-    result rows to the levels above
-  - the executor uses the top node's `Next()` results to construct the final
-	result set to return to the client.
-- `Start()` is meant to run execution steps prior to the creation of
-  the first result row. Some nodes (e.g. `deleteNode()`) can do all
-  their work in `Start()`.
-- *can run all statements*
-- fully sequential execution per statement
-
-From a software engineering perspective, it's a little weird to make
-`Start()` and `Next()` *methods* of the data structure they are really
-consuming as input (the logical plan) but not otherwise
-modifying. It's also weird and rather unfortunate that the internal
-execution state of the local runner is also carried as attributes of
-the logical plan, because this means the logical plan nodes must
-allocate space for it even when the local runner is not used.
-
-These shortcomings are an artifact from an earlier time where there
-was no conceptual separation between constructing the logical plan and
-running it, and *all* operations using `planNode`s were implemented as
-`planNode` methods.  We're moving (slowly) away from that; eventually
-`Next()` and `Start()` will be detached too and the local runner's
-internal state will sit in separate data structures.
-
-**Code:** the `Start()` and `Next()` methods of all `planNode`s.
+**Interface:** see `opt`.
 
-**Whom to ask for details:** the authors of the respective methods.
+**Whom to ask for details:** the optimizer team.
 
-### SQL Expression evaluator
+## SQL back-end
 
-This is conceptually a part of the local runner, but is well
-encapsulated in the `Eval()` method of `Expr` nodes and is used
-independently by distSQL processors, so it deserves its own
-sub-section here.
+### Physical planner and distributed execution
 
-**Role:** `TypedExpr` in, `Datum` out.
+**Role:** prepare the distribution of query execution and then
```

Comment:
does "prepare" here mean something distinct from "plan"? "plan" seems to me it would be more consistent and also hint at the eventual desire to move this up the stack into the optimizer